tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -3133,11 +3193,23 @@ ReductionCodegenInfo IrEmitterUnnested::ComputeReductionCodegenInfo(
     return kWarpSize;
   }();
 
+  int tile_size_x = reduction_tiling[2] * num_threads_x;
+
+  int vector_size = 1;
+  if (indexing_order == kLinearStridedIndexingX) {
+    if (reduction_dimensions.dimensions[2] % 2 == 0 &&
+        // Assuming XLA will perform the unrolling and LLVM will vectorize,
+        // disable the unroll for case that LLVM doesn't vectorize.
+        !MayPreventVectorization(*unnested_hlo)) {
```

Comment:
Is this why the change to `MayPreventVectorization` was necessary? Or is that for fused computations as well?